<launch>
    <arg name="bag_file" default="" />
    <arg name="camera_file" default="$(find omni_slam_eval)/launch/default_camera.yaml" />
    <arg name="results_file" default="$(eval ''.join(arg('bag_file').split('.')[:-1]) + '.' + ''.join(arg('camera_file').split('/')[-1].split('.')[:-1]) + '.odometry.hdf5')" />
    <arg name="rate" default="1" />
    <node pkg="omni_slam_eval" type="omni_slam_odometry_eval_node" name="omni_slam_odometry_eval_node" required="true" output="screen">
        <param name="bag_file" value="$(arg bag_file)" />
        <param name="results_file" value="$(arg results_file)" />
        <param name="image_topic" value="/unity_ros/Camera/FisheyeCamera/image_raw" />
        <param name="depth_image_topic" value="/unity_ros/Camera/FisheyeDepthCamera/image_raw" />
        <param name="pose_topic" value="/unity_ros/Camera/TrueState/pose" />
        <param name="tracked_image_topic" value="/omni_slam/tracked" />
        <param name="odometry_estimate_topic" value="/omni_slam/odometry" />
        <param name="odometry_optimized_topic" value="/omni_slam/odometry_optimized" />
        <param name="odometry_ground_truth_topic" value="/omni_slam/odometry_truth" />
        <param name="path_estimate_topic" value="/omni_slam/odometry_path" />
        <param name="path_optimized_topic" value="/omni_slam/odometry_path_optimized" />
        <param name="path_ground_truth_topic" value="/omni_slam/odometry_path_truth" />
        <param name="output_frame" value="map" />
        <param name="rate" value="$(arg rate)" />
        <rosparam subst_value="true">
            detector_type: 'GFTT'
            detector_parameters: {maxCorners: 2000, qualityLevel: 0.001, minDistance: 5, blockSize: 5}
            keyframe_interval: 1
            tracker_type: 'lk'
            tracker_window_size: 128
            tracker_num_scales: 4
            tracker_checker_epipolar_threshold: 0.008
            tracker_checker_iterations: 1000
            tracker_delta_pixel_error_threshold: 0.0
            tracker_error_threshold: 20.0
            tracker_template_update_rate: 1
            min_features_per_region: 100
            max_features_per_region: 1000
            odometry_type: 'pnp'
            pnp_inlier_threshold: 3.0
            pnp_iterations: 3000
            bundle_adjustment_max_iterations: 1000
            bundle_adjustment_loss_coefficient: 0.05
            bundle_adjustment_logging: true
            bundle_adjustment_num_threads: 20
        </rosparam>
    </node>
    <rosparam command="load" file="$(arg camera_file)" ns="omni_slam_odometry_eval_node" />
</launch>

